### PR TITLE
Imperative interpreter and tests

### PR DESCRIPTION
**Pull request by benozol**
Sorry I failed to branch in time and this PR targets #5 and #6. The changes are

- fully imperative concrete interpreter
- tests for the interpreter `tests/*.cls`
-  `./run_tests.sh`, that compares the tests with `tests/*.oracle` using.

### CODE REVIEW

File: src/concrete/concrete_colis.ml [hunk 1]
```diff
@@ -56,65 +57,80 @@ let parse_args () =
...
+
 (** Print execution results *)
-let print_result behaviour value stdout =
-  printf "BEHAVIOUR: %s@." behaviour;
-  printf "VALUE: %b@." value;
+let print_outcome outcome =
+  printf "RESULT: %b@." outcome.result;
   printf "STDOUT:@.";
-  List.iter (printf "%s@\n") (List.rev stdout)
+  List.iter (printf "> %s@\n") (List.rev outcome.stdout)
```
Comment (Niols) [inline src/concrete/concrete_colis.ml]: Actually, considering that the "behaviour" of a "program" value is a boolean and that the only other thing is a standard output, why don't we just have an interpreter printer on stdout and exiting with 0 or 1? (>1 being for errors)

**Comment by Niols** [comment]
I couldn't look at all of it, but it looks nice. I'm not sure that I like the way the oracle system works. Is there more in an oracle than a behaviour and a standard output? If no, then why do we have to care about oracles in the project when it should only be something like a diff in the test suite?

**Comment by Niols** [comment]
I suggest we keep it that way for now. I would like to merge this PR, then merge #8 into master (I can take care of the merges) so that we have a single consistent branch to show in 45 minutes. What do you say @benozol?

File: src/concrete/concrete_colis.ml [hunk 2]
```diff
@@ -1,12 +1,13 @@
...
 open Scanf
-open Semantics__Context
-open Interpreter__Interpreter
+module Context = Semantics__Context
+module Buffers = Semantics__Buffers
+module Interpreter = Interpreter__Interpreter
 
 (** Parse program arguments.
 
     Replace with Cmdliner or so?
```
[the remark this answers is not in the document]
Comment (benozol) [reply src/concrete/concrete_colis.ml]: Yes please! Args is no fun.

[the remark this answers is not in the document]
Comment (Niols) [reply src/concrete/concrete_colis.ml]: I put that in #14 so I can resolve this conversation for this PR :-)

**Comment by benozol** [comment]
I am working on it. I had to squash your commits from #8 due to (instruction/statement confusion) and have a broken test that I didn’t notice before. Therefore, I shared the preliminary merge of #8 and #13 in: #15

**Pull request closed without merging**